src/CppUTestExt/MockSupport_c.cpp

```diff
@@ -382,7 +384,7 @@ MockExpectedCall_c* withConstPointerParameters_c(const char* name, const void* v
 
 MockExpectedCall_c* withFunctionPointerParameters_c(const char* name, void (*value)())
 {
-    expectedCall = &expectedCall->withParameter(name, value);
+    expectedCall = &expectedCall->withParameter(name, (Cpp_Func*)value);
```

Comment:
At least (void (*)(void)) compiled without any problem. And this cast is in C++ context. Could you try it for CL2k?